test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+    const [ns] = names;
+
+    const nameInfo = await wallet.getName(ns.name);
+
+    assert.deepEqual(ns, nameInfo);
+  });
+
+  it('should fail to open a bid without a bid value', async () => {
+    const fn = async () => (await wallet.createBid({
+      name: name
+    }));
+
+    assert.rejects(fn);
+  });
+
+  it('should fail to open a bid without a lockup value', async () => {
+    const fn = async () => (await wallet.createBid({
+      name: name,
+      bid: 1000
+    }));
+
+    assert.rejects(fn);
+  });
+
+  it('should create a reveal', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const {info} = await nclient.execute('getnameinfo', [name]);
+    assert.equal(info.name, name);
+    assert.equal(info.state, 'REVEAL');
+
+    const json = await wallet.createReveal({
+      name: name
+    });
+
+    const reveals = json.outputs.filter(output => output.covenant.type === types.REVEAL);
+    assert.equal(reveals.length, 1);
+  });
+
+  it('should get auction info', async () => {
+    const names = await wallet.getNames();
+
+    await sleep(100);
+
+    assert(names.length > 0);
+    const [,ns] = names;
+
+    const auction = await wallet.getAuction(ns.name);
+
+    // auction info returns a list of bids
+    // and a list of reveals for the name
+    assert.ok(auction.bids);
+    assert.ok(auction.reveals);
+  });
+
+  it('should create a redeem', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    // wallet2 wins the auction, wallet can submit redeem
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    await wallet2.createBid({
+      name: name,
+      bid: 2000,
+      lockup: 3000
+    });
+
+    await sleep(100);
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createReveal({
+      name: name
+    });
+
+    await wallet2.createReveal({
+      name: name
+    });
+
+    const {revealPeriod} = network.names;
+    for (let i = 0; i < revealPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const json = await wallet.createRedeem({
+      name: name
+    });
+
```

Comment:
I could be reading this wrong, but `wallet` had the LOWER bid, right? So shouldn't this fail? Shouldn't `wallet2` be able to succeed in creating a `REDEEM`? I'm sure this is checked in other tests, but I dunno if you also want to assert that the redeem values are correct (winning bid / 2nd-highest bid / change value)

Reply:
`REDEEM` is for the loser to get their money back. There should be an error if the winner tries to submit a `REDEEM` so this raises a good point around an additional test case